app/javascript/controllers/application_controller.js

```diff
@@ -1,18 +1,34 @@
 import { Controller } from 'stimulus'
 import StimulusReflex from 'stimulus_reflex'
-import { setFocus } from '../src/focus'
 
 export default class extends Controller {
   connect () {
     StimulusReflex.register(this)
   }
 
-  beforeReflex () {
-    document.getElementById('reflex-status').classList.remove('d-none')
+  beforeReflex (element) {
+    document
+      .querySelectorAll('.reflex-activity')
+      .forEach(e => (e.hidden = false))
   }
 
-  afterReflex () {
-    document.getElementById('reflex-status').classList.add('d-none')
-    setFocus()
+  afterReflex (element) {
+    this.focus(element)
+  }
+
+  shouldFocus (element) {
+    if (document.activeElement === element) return false
+    if (element.tagName === 'TEXTAREA') return true
+    if (element.tagName !== 'INPUT') return false
```

Comment:
we may want to add the date field input here. Until we add one of our date pickers, trying to type in a date on safari causes the focus issue.

Reply:
I think we can punt on that for now as a date picker will soon follow.